File: core/src/ionium/util/MemoryUtils.java
package ionium.util;

public class MemoryUtils {

	private MemoryUtils() {

	}

	public static int getUsedMemory() {
		return (int) (Runtime.getRuntime().totalMemory() / 1024);
	}

	public static int getMaxMemory() {
		return (int) (Runtime.getRuntime().maxMemory() / 1024);
	}

	public static int getFreeMemory() {
		return (int) (Runtime.getRuntime().freeMemory() / 1024);
	}

	public static int getCores() {
		return Runtime.getRuntime().availableProcessors();
	}
}

File: core/src/main/kotlin/io/github/chrislo27/toolboks/util/MemoryUtils.kt
package io.github.chrislo27.toolboks.util

/**
 * Some properties you can use to get the current memory stats *in kilobytes*.
 */
object MemoryUtils {

    val usedMemory: Int
        get() = (Runtime.getRuntime().totalMemory() / 1024).toInt()

    val maxMemory: Int
        get() = (Runtime.getRuntime().maxMemory() / 1024).toInt()

    val freeMemory: Int
        get() = (Runtime.getRuntime().freeMemory() / 1024).toInt()

}